### DISCUSSION TITLE
adding a label to a pretrained textcat

### DISCUSSION THREAD

**Discussion by mathiasdus** [explosion/spaCy#10144] [Help: Coding & Implementations]
Hello,

I know that since v3 of Spacy, textcatBow is resizable. Hence, adding a label to an already trained texcat_multilabel pipeline should not be a problem. However, when I call textcat.add_label(), I get the error: 
```[E922] Component 'textcat_multilabel' has been initialized with an output dimension of 30 - cannot add any more labels.```
 
Why is this happening?

**Comment by adrianeboyd** [comment]
Resizing for textcat was added in v3.1.0. Can you check which version of spacy you're using (`spacy info`) and whether the textcat architectures in the config correspond to `spacy.TextCatBOW.v2` or `spacy.TextCatCNN.v2`? If they're the `v1` versions, you would need to retrain the model from scratch with the newer architectures in order to be able to add new labels.
